index.js

```diff
@@ -1,18 +1,90 @@
 const readline = require('readline');
 const rl = readline.createInterface(process.stdin, process.stdout);
+// Global Variables Assigned Here
+let answer
 
+// Promise Function
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
-
-start();
-
+// Return Random Number Funciton
+function randomNum(min, max) {
+  return Math.floor(Math.random() * (max - min + 1) + min)
+}
+// Return Mid Value Function (whole number)
+function midValue(min, max) {
+  return Math.floor((min + max) / 2)
+}
+// Start Game Function
+start()
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  console.log('Welcome to my Guessing Game!\n' + '*Games alternate after intital selection*\n')
+  let gameChoice = await ask('To Start Please Choose Either Computer Guesses or Human Guesses\n')
+  gameChoice = gameChoice.trim().toLowerCase()
+  if (gameChoice == 'computer guesses') {
+    computerGuesses()
+  }
+  if (gameChoice == 'human guesses') {
+    humanGuesses()
+  }
 }
+// Computer Guess Game Function
+async function computerGuesses() {
+  console.log('\nYou have now entered Mode: Computer Guesses\n' + 'To play this game please choose a numerical range for the game(ex. 1 to 100)')
+  let userMinRange = parseInt(await ask('What is your desired low end range?\n'));
+  let userMaxRange = parseInt(await ask('What is your desired top end range?\n'));
+  console.log('You entered: ' + userMinRange + ' to ' + userMaxRange)
+  let computerGenNum = randomNum(userMinRange, userMaxRange)
+  console.log("Please choose a secret number within your chosen range now and remember it")
+  let answer = await ask('Is your number ' + computerGenNum + ' (yes / no)' + '?\n')
+
+  while (answer !== 'yes') {
+    let highLow = await ask('Is your number lower or higher or correct?\n');
+    highLow = highLow.trim().toLowerCase()
+    if (highLow === 'higher') {
+      userMinRange = computerGenNum + 1
+      computerGenNum = midValue(userMinRange, userMaxRange)
+      console.log('My next guess is ' + computerGenNum + '\n')
+    } else if (highLow === 'lower') {
+      userMaxRange = computerGenNum - 1
+      computerGenNum = midValue(userMinRange, userMaxRange)
+      console.log('My next guess is ' + computerGenNum + '\n')
+    } else if (highLow === 'correct') {
+      console.log('I have Won!\n' + 'Skynet Activated')
+      humanGuesses()
+    } else {
+      console.log('Unrecognized input, Try Again')
+    } if (userMaxRange < userMinRange) {
+      console.log('Cheater! Start Over')
```

Comment:
Nice cheat-detection! Small criticism, but immediately before calling the player out for cheating, the program `console.log()`s a value outside of the previously defined range.  It will say, `My next guess is 0` (0 being outside the given range) followed immediately by `Cheater! Start over`.  Consider re-ordering the conditions so that final guess is not triggered, but rather immediately calls out the player for cheating.